zeebe/journal/src/main/java/io/camunda/zeebe/journal/JournalInfoProvider.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.journal;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface JournalInfoProvider {
+  CompletableFuture<Segments> getTailSegments(long index);
```

Comment:
:question: Why pass a specific type instead of `Map<Long, Path>`? Do you expect the `Segments` type to grow in the future, or perform some kind of functionality?

Reply:
When I first wrote I was tempted to add `committed_index` for example, so I thought it could be used in the future. As of now it's potentially overcomplicating things, but we were already discussing about addding an optimization related to the commit index. 
The name `Segments` is probably misleading though.

Reply:
I think it's better for now to just return a `Collection<Path>` as you suggest.